addon/v1/types.go

```diff
@@ -0,0 +1,178 @@
+	// +required
+	Resource string `json:"resource"`
+	// name of the referent.
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+}
+
+type ConditionStatus string
+
+// These are valid condition statuses. "ConditionTrue" means a resource is in the condition.
+// "ConditionFalse" means a resource is not in the condition. "ConditionUnknown" means kubernetes
+// can't decide if a resource is in the condition or not. In the future, we could add other
+// intermediate conditions, e.g. ConditionDegraded.
+const (
+	ConditionTrue    ConditionStatus = "True"
+	ConditionFalse   ConditionStatus = "False"
+	ConditionUnknown ConditionStatus = "Unknown"
+)
+
+// AddonStatusCondition represents the state of the operator's
+// managed and monitored components.
+// +k8s:deepcopy-gen=true
+type AddonStatusCondition struct {
+	// type specifies the aspect reported by this condition.
+	// +kubebuilder:validation:Required
+	// +required
+	Type AddonStatusConditionType `json:"type"`
+
+	// status of the condition, one of True, False, Unknown.
+	// +kubebuilder:validation:Required
+	// +required
+	Status ConditionStatus `json:"status"`
+
+	// lastTransitionTime is the time of the last update to the current status property.
+	// +kubebuilder:validation:Required
+	// +required
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+
+	// reason is the CamelCase reason for the condition's current status.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// message provides additional information about the current condition.
+	// This is only to be consumed by humans.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+// Release represents an OpenShift release image and associated metadata.
+// +k8s:deepcopy-gen=true
+type Release struct {
+	// version is a semantic versioning identifying the update version. When this
+	// field is part of spec, version is optional if image is specified.
+	// +required
+	Version string `json:"version"`
+
+	// imageManifest is list of images will be used for the specific version
+	// this will be use for backward compatability (i.e hub upgraded but agent have not)
+	// +required
+	ImageManifest []Image `json:"imageManifest"`
+}
+
+type Image struct {
+	Name  string `json: "name"`
+	Image string `json: "image"`
+}
+
+// AddonStatusConditionType is an aspect of operator state.
+type AddonStatusConditionType string
+
+const (
+	// Available indicates that the operand (eg: openshift-apiserver for the
+	// openshift-apiserver-operator), is functional and available in the cluster.
+	Available AddonStatusConditionType = "Available"
+
+	// Progressing indicates that the operator is actively rolling out new code,
+	// propagating config changes, or otherwise moving from one steady state to
+	// another.  Operators should not report progressing when they are reconciling
+	// a previously known state.
+	Progressing AddonStatusConditionType = "Progressing"
```

Comment:
Given the combination of clusteroperator and clusterversion here, I would expect to see an `Upgrading` condition type too.